Modules/Core/Common/test/itkCommonTypeTraitsGTest.cxx

```diff
@@ -84,3 +84,30 @@ TEST(CommonTypeTraits, ContinuousIndexIsPOD) {
   EXPECT_EQ(std::is_trivial<T>::value, true);
   EXPECT_EQ(std::is_standard_layout<T>::value, true);
 }
+
+/************ FixedArray: noexcept move checks *************/
+
+/* Dummy class without noexcept move contructors. */
+struct MightThrowAtMove
+{
+    MightThrowAtMove() {}
+    MightThrowAtMove(MightThrowAtMove&&) {}
+    MightThrowAtMove(const MightThrowAtMove&) {}
+};
+
+/* Check that FixedArray works as an agregate class.
+ * Move constructors are noexcept when the contained class has a noexcept move constructor.
+ * And when the contained class might throw at move, the qualifer automatically
+ * propagates it to FixedArray.
+ * This is the same behaviour than std::array. It proves there is no need to
+ * add noexcept (ITK_NOEXCEPT) to the container. This would disallow using it
+ * with might-throw-objects.
+ */
+TEST(CommonTypeTraits, FixedArrayIsNoExceptMovable) {
+  using TInt = itk::FixedArray<int, 3>;
+  EXPECT_EQ(std::is_nothrow_move_constructible<TInt>::value, true);
+  EXPECT_EQ(std::is_copy_constructible<TInt>::value, true);
+  using TMightThrowAtMove = itk::FixedArray<MightThrowAtMove, 3>;
```

Comment:
Maybe you should rename `TInt` and `TMightThrowAtMove` to something like "IntArrayType" and "MightThrowAtMoveArrayType". Because I _think_ the `T` prefix was meant for _formal_ template parameters rather than for the actual types.

Reply:
I agree, thanks.